test/e2e/e2e_test.go

```diff
@@ -3,38 +3,59 @@
 package e2e
 
 import (
-	"flag"
-	"fmt"
 	"os"
 	"testing"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 
+	"github.com/Microsoft/ApplicationInsights-Go/appinsights"
 	"github.com/sirupsen/logrus"
 
+	"github.com/openshift/openshift-azure/pkg/util/log"
 	_ "github.com/openshift/openshift-azure/test/e2e/specs"
 	_ "github.com/openshift/openshift-azure/test/e2e/specs/fakerp"
 	_ "github.com/openshift/openshift-azure/test/e2e/specs/realrp"
 	"github.com/openshift/openshift-azure/test/reporters"
+	"github.com/openshift/openshift-azure/test/sanity"
 )
 
 var (
 	gitCommit = "unknown"
 )
 
 func TestE2E(t *testing.T) {
-	fmt.Printf("e2e tests starting, git commit %s\n", gitCommit)
-
-	flag.Parse()
-	logrus.SetFormatter(&logrus.TextFormatter{})
-	logrus.SetOutput(GinkgoWriter)
-	logrus.SetReportCaller(true)
+	icli := appinsights.NewTelemetryClient(os.Getenv("AZURE_APP_INSIGHTS_KEY"))
+	icli.Context().CommonProperties["type"] = "ginkgo"
+	icli.Context().CommonProperties["resourcegroup"] = os.Getenv("RESOURCEGROUP")
+
+	logger := os.Stdout
```

Comment:
you should probably use logrus' testing log/capture facilities instead: https://github.com/sirupsen/logrus#testing

Reply:
This is for hooking into `os.Stdout` file descriptor. We basically hooking up into filedescriptor for os.stdout which is being used for ginkoWriter.